tests/manipulation/test_pixel_manipulation.py

```diff
@@ -1,10 +1,79 @@
-def test_get_stratified_coords():
-    # TODO test no out of range in coords
-    # TODO test randomness (masked pixel distribution doesn't show any pattern)
+import numpy as np
+import pytest
 
-    # coordinate_grid += grid_random_increment outputs oor values
-    pass
+from careamics_restoration.manipulation.pixel_manipulation import (
+    default_manipulate,
+    get_stratified_coords,
+)
 
 
-def tesst_defaul_manipulate():
-    pass
+@pytest.mark.parametrize(
+    "mask_pixel_perc, shape, num_iterations",
+    [(0.4, (32, 32), 1000), (0.4, (10, 10, 10), 1000)],
+)
+def test_get_stratified_coords(mask_pixel_perc, shape, num_iterations):
+    """Test the get_stratified_coords function.
+
+    Ensure that the array of coordinates is randomly distributed across the
+    image and doesn't demonstrate any strong pattern.
+    """
+    # Define the number of iterations to run the test and the array
+    array = np.zeros(shape)
+
+    # Iterate over the number of iterations and add the coordinates
+    for _ in range(num_iterations):
+        # Calculate distance between masked pixels as it's done in the
+        # get_stratified_coords function
+        coords = get_stratified_coords(mask_pixel_perc, shape)
+        # Check that the coordinates are within the shape of the array
+        # Check that the distance between masked pixels is approximately the same,
+        # and distance from border is equal
+        for coord_pair in coords:
+            assert len(coord_pair) == len(shape)
+            assert all(coord_pair) >= 0
+            assert [c <= s for c, s in zip(coord_pair, shape)]
+
+        array[tuple(np.array(coords).T.tolist())] += 1
+
+    # Check that the maximum value of the array is less than half of the second
+    # largest. This is to ensure that there's no strong pattern in the array.
+    hist = sorted(np.histogram(array, bins=100)[0])
```

Comment:
I would compare the minimum with the maximum. The smaller the distance between the two, the more homogeneous.

And I would mention that these numbers are directly proportional to the probability of being a masked pixel.

If there are 0 values, we could count them and have a threshold for failure. For instance we accept a number of 0 values equal to the sum of all spatial dimensions (= we accept about one pixel line per dimension). Let's see if that pass!

Reply:
@jdeschamps Could you provide some pseudocode?

Reply:
Ok, given that it will depend heavily on the shape and pix_perc I am not sure there is a good way to have a threshold on the probability...

Could you share with me your test notebook? I'd like to play around with it.

For now, maybe just check the number of 0 values with heuristic value:
```python
# Check that the number of unselectable pixels is less than about a line in each dimension
assert sum(array == 0) < np.sum(shape)
```